### PR TITLE
Modifying RLMSupport.swift to support both Swift 2.2 and Swift 3

### PR DESCRIPTION
**Pull request by austinzheng**
@bdash @tgoyne @jpsim

### CODE REVIEW

File: Realm/Swift/RLMSupport.swift [hunk 1]
```diff
@@ -18,57 +18,114 @@
...
+    extension RLMObject {
+        // Swift query convenience functions
+        public class func objectsWhere(predicateFormat: String, _ args: CVarArg...) -> RLMResults<RLMObject> {
+            return objects(with: NSPredicate(format: predicateFormat, arguments: getVaList(args)))
+        }
 
-    public class func objectsInRealm(realm: RLMRealm, _ predicateFormat: String, _ args: CVarArgType...) -> RLMResults {
-        return objectsInRealm(realm, withPredicate:NSPredicate(format: predicateFormat, arguments: getVaList(args)))
+        public class func objectsInRealm(realm: RLMRealm, _ predicateFormat: String, _ args: CVarArg...) -> RLMResults<RLMObject> {
+            return objects(in: realm, with:NSPredicate(format: predicateFormat, arguments: getVaList(args)))
```
Comment (jpsim) [inline Realm/Swift/RLMSupport.swift]: missing space: `with: NSPredicate`

File: Realm/Swift/RLMSupport.swift [hunk 2]
```diff
@@ -18,57 +18,114 @@
...
+    extension RLMObject {
+        // Swift query convenience functions
+        public class func objectsWhere(predicateFormat: String, _ args: CVarArgType...) -> RLMResults {
+            return objectsWithPredicate(NSPredicate(format: predicateFormat, arguments: getVaList(args)))
+        }
 
-    public func objectsWhere(predicateFormat: String, _ args: CVarArgType...) -> RLMResults {
-        return objectsWithPredicate(NSPredicate(format: predicateFormat, arguments: getVaList(args)))
+        public class func objectsInRealm(realm: RLMRealm, _ predicateFormat: String, _ args: CVarArgType...) -> RLMResults {
+            return objectsInRealm(realm, withPredicate:NSPredicate(format: predicateFormat, arguments: getVaList(args)))
```
Comment (jpsim) [inline Realm/Swift/RLMSupport.swift]: missing space: `with: NSPredicate`

File: Realm/Swift/RLMSupport.swift [hunk 3]
```diff
@@ -18,57 +18,114 @@
...
+        public class func objectsInRealm(realm: RLMRealm, _ predicateFormat: String, _ args: CVarArg...) -> RLMResults<RLMObject> {
+            return objects(in: realm, with:NSPredicate(format: predicateFormat, arguments: getVaList(args)))
+        }
     }
-}
 
-public final class RLMIterator: IteratorProtocol {
-    private let generatorBase: NSFastGenerator
+    public final class RLMIterator: IteratorProtocol {
+        private let generatorBase: NSFastEnumerationIterator
```
Comment (jpsim) [inline Realm/Swift/RLMSupport.swift]: `iteratorBase`?

**Comment by jpsim** [comment]
👍  (PS: mark PRs with `S:Review` when they're ready for review)

**Comment by austinzheng** [comment]
@jpsim Do we run swiftlint on the RLMSupport file?

**Comment by jpsim** [comment]
> Do we run swiftlint on the RLMSupport file?

no, but we *could*:

```diff
-included: RealmSwift-swift2.2
+included:
+  - RealmSwift-swift2.2
+  - Realm/Swift
```

Just have 2 line length violations at the moment.

**Pull request merged**